pipeline_dp/dataset_histograms/histograms.py

```diff
@@ -66,7 +77,15 @@ class HistogramType(enum.Enum):
 
 @dataclass
 class Histogram:
-    """Represents a histogram over integers."""
+    """Represents a histogram over integers.
+
+    To calculate range of the histogram (i.e. min and max values on axis X)
```

Comment:
Let's introduce definitions of histograms:

There are 2 types of histograms: 
1. Integer histogram, which is used for counts and bins are spaced logarithmically (see `computing_histograms. _to_bin_lower_logarithmic` for details how bins are generated )
2. Floating histograms, which is used for sums per partition. Which has the same size bins.

Please introduce property `IsInteger`  for detecting histogram type (this property can be evalualted based on Histogram.name)

Reply:
added, wdyt? Not sure, maybe we should make it the property of the `HistogramType` enum and not the property of Histogram class? 
and whether it should be bool. Better maybe to make it another enum (e.g. HistogramValueType) or return just `type(bins[0].lower)` or put as property of `HistogramType` enum (e.g. `HistogramType.L0_CONTRIBUTIONS.value_type` will return `int`) and then implement `lower` and `upper` accordingly.

I think "`HistogramType.L0_CONTRIBUTIONS.value_type` will return `int`" is the best interface.

Reply:
About enum, it might be a good a solution, but for me it's not that clear how excatly it will evolve. E.g. we might want to have histogram computed in different way - uniform (as sum) or logarithmic (as others) and maybe others type of logarithmic. But it's not that clear. So I'd say having bool property is good, because it's simple. 

The same with type of `HistogramType`, we don't know how it will evolve, in principle we might want to have different of histogram generation for the same type. Property of `Histogram` seems to be good and simple default.

`type(bins[0].lower)` is not reliable enough in Python, since Python doesn't have static type system, and in principle that it's double/int, bins[0].lower is int/float is not implementation dependent.

Reply:
I see. implemented, wdyt?